File: programmers/codingtest_practice/gcd_and_lcm/GcdAndLcm.java
package programmers.codingtest_practice;

public class GcdAndLcm {

	public static int[] solution(int n, int m) {
		int[] answer = {};
		answer = new int[2];
		
		int min = 0;
		if(n > m) min = m;
		else min = n;
		
		for(int i=min; i>=1; i--) {
			if(n % i == 0 && m % i == 0) {
				answer[0] = i;
				break;
			}
		}
		
		answer[1] = n * m / answer[0];
		
		return answer;
	}
	
	public static void main(String[] args) {
		solution(3, 12);
	}
}

File: programmers/코딩 기초 트레이닝/gcd_and_lcm/gcd_and_lcm.py
def solution(n, m):
    big, small, div = (m, n, n) if n < m else (n, m, m)
    answer = [1,1]
    while div != 1 :
        if small % div == 0 and big % div == 0 :
            big, small = big/div, small/div
            answer[0] *= div
            div = small
        else :
            div -= 1
    answer[1] = big*small*answer[0]
    return answer